### ISSUE TITLE
Issues removing and updating box additions with Virtualbox 4.3

### ISSUE THREAD

**Issue by trkoch** [dotless-de/vagrant-vbguest#95]
I'm trying to upgrade the guest box additions of a Ubuntu Precise box from 4.2 to 4.3 after upgrading Virtualbox (4.3.0) and Vagrant (1.3.5). The plugin correctly detects a version mismatch, installs dependencies and kicks of the installer. However, the installer quits with a warning that a previously version is still installed.

Manually installing works (with exception of the X11 additions, but I guess that's to be expected), when rebooting Vagrant still detects the old 4.2 additions though.

https://gist.github.com/trkoch/7197144

**Comment by tgr** [comment]
Recently I see the `vboxadd` error a lot:
```
==> default: Machine booted and ready!
GuestAdditions seems to be installed (4.3.10) correctly, but not running.
bash: line 3: /etc/init.d/vboxadd: No such file or directory
bash: line 3: /etc/init.d/vboxadd: No such file or directory
==> default: Checking for guest additions in VM...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

/etc/init.d/vboxadd setup

Stdout from the command:



Stderr from the command:

bash: line 3: /etc/init.d/vboxadd: No such file or directory
```

The box is useless as folder sharing does not work.

The effect is somewhat random, about one in ten times vagrant displays the same error message but then pulls the box up fine:
```
==> default: Machine booted and ready!
GuestAdditions seems to be installed (4.3.10) correctly, but not running.
bash: line 3: /etc/init.d/vboxadd: No such file or directory
==> default: Checking for guest additions in VM...
```

**Comment by andrelohmann** [comment]
I updated my ubuntu box with virtualbox-5.0 from the oracle repositories and get the same error message since then.

Has something changed in vbox 5, so that /etc/init.d/vboxadd isn't available any longer and does an alternativity exists, that can be implemented into vagrant-vbguest?

**Comment by ajorpheus** [comment]
A combination of the advice above worked for me:

1. Get the name of the Virtualbox VM you are interested in:
```/Applications/VirtualBox.app/Contents/MacOS/VBoxManage list vms```
(Assume that the name is ```blah```)

2. Reset the version recorded in the Virtualbox VM:
```/Applications/VirtualBox.app/Contents/MacOS/VBoxManage guestproperty set "devops-development-environment" /VirtualBox/GuestAdd/Version```

3. Confirm that the version has been wiped out
```/Applications/VirtualBox.app/Contents/MacOS/VBoxManage guestproperty get "devops-development-environment" /VirtualBox/GuestAdd/Version```
(Should get 'No Value Set!')

4. SSH inside the VM and uninstall existing guest additions and reboot:
```ssh -p 2200 vagrant@localhost;
  /opt/VirtualBoxGuestAdditions/uninstall.sh;
  rm -rf /tmp/Virtualbox;
  sudo reboot;```

5. Install it again:
```﻿VAGRANT_LOG=info vagrant vbguest --do install```

6. Finally check the status
```vagrant vbguest --status```
> GuestAdditions 5.0.10 running --- OK.

**Comment by tobias-dev** [comment]
Whatever I tried I still couldn't get rid of the following error when running `vagrant up`:

```
==> default: Machine booted and ready!
GuestAdditions seems to be installed (5.0.10) correctly, but not running.
Usage: service < option > | --status-all | [ service_name [ command | --full-restart ] ]
bash: line 4: vboxadd: command not found
```

So, the VirtualBox Guest Additions somehow failed to start. I ended up creating a small start script at `/etc/init/vboxadd.conf` with the following content and that work for me:

```
description "VirtualBox Linux Additions"

start on runlevel [2345]
stop on runlevel [!2345]

respawn
exec /sbin/rcvboxadd start
```

**Comment by jeffgeorge** [comment]
@tobias-dev: Your first recommendation (creating a new startup script) worked for me, but it looks like it might be a *bit* aggressive. I was seeing it *constantly* trying to restart Guest Additions all throughout boot and beyond, resulting in Virtualbox gobbling up 20% of my host's CPU Utilization while *idle*.

Your second recommendation of just symlinking `vboxadd` / `rcvboxadd` seems to be a perfect workaround, though. I still get the underlying issue reported above of vbguest being unable to determine what version of Virtualbox is installed on my host, but at least now it's *working*.

**Comment by Fuxy22** [comment]
I manage to fix mine by issuing as mentione by @trkoch 

**1.** `/usr/bin/VBoxManage guestproperty set <uid> /VirtualBox/GuestAdd/Version <version>`

that got rid of 
```
Got different reports about installed GuestAdditions version:
Virtualbox on your host claims:   5.0.6
VBoxService inside the vm claims: 5.0.10
Going on, assuming VBoxService is correct...
```

**2.** SSH-ing into the box and issuing `sudo ln -s /sbin/rcvboxadd /sbin/vboxadd`

as suggested by @tobias-dev that got rid of the
```
bash: line 4: vboxadd: command not found
```

Now if I could only figure out why it's issuing the wrong **service** command i could be golden.

```
==> default: Machine booted and ready!
GuestAdditions seems to be installed (5.0.10) correctly, but not running.
Usage: service < option > | --status-all | [ service_name [ command | --full-restart ] ]
Starting the VirtualBox Guest Additions ...done.
```
P.S. I'm running Laravel Homestead which is a Ubuntu 14.04.3 LTS box.

**Comment by barefootcoder** [comment]
I am also seeing these two errors on startup:
```
Usage: service < option > | --status-all | [ service_name [ command | --full-restart ] ]
bash: line 4: vboxadd: command not found
```
However, I don't see anything failing to mount, or really any failures at all.  But they bug me, so I went looking for a solution and here I am.

I do not have an `/sbin/rcvboxadd` to symlink, but, by using `locate`, I was able to find an alternative:
```
sudo ln -s /opt/VBoxGuestAdditions-4.3.18/init/vboxadd /sbin
```
Like Fuxy22, this got rid of the second error, but not the first.  I can't seem to find whatever script is causing this to fix the borked `service` command.  Any help would be appreciated.

Hopefully this helps debug the issue.

**Comment by NeMO84** [comment]
I got this working similar to @ajorpheus. 

The guest property of the host version is never kept in sync with version upgrades of VirtualBox on my machine. Even though the guest machine reports the latest version installed correctly. 

So I wrote a quick script to help me maintain the version upon upgrades more easily: https://gist.github.com/NeMO84/95e93346e81dbb2b75dd677922addbda

Hope it helps others. We still need a long term solution for
